pandora/models/pandora.py

```diff
@@ -284,37 +284,99 @@ class BookmarkList(PandoraModel):
 
 class SearchResultItem(PandoraModel):
 
-    artist = Field("artistName")
-    song_name = Field("songName")
     score = Field("score")
-    likely_match = Field("likelyMatch", default=False)
     token = Field("musicToken")
 
     @property
     def is_song(self):
-        return self.token.startswith('S')
+        return type(self) is SongSearchResultItem
 
     @property
     def is_artist(self):
-        return self.token.startswith('R')
+        return type(self) is ArtistSearchResultItem and \
+               self.token.startswith("R")
 
     @property
     def is_composer(self):
-        return self.token.startswith('C')
+        return type(self) is ArtistSearchResultItem and \
+               self.token.startswith("C")
+
+    @property
+    def is_genre_station(self):
+        return type(self) is GenreStationSearchResultItem
 
     def create_station(self):
-        if self.is_song:
-            self._api_client.create_station(track_token=self.token)
+        raise NotImplementedError
+
+    @classmethod
+    def from_json(cls, api_client, data):
+        if data["musicToken"].startswith("S"):
+            return SongSearchResultItem.from_json(api_client, data)
+
+        elif data["musicToken"].startswith("R") or \
+                data["musicToken"].startswith("C"):
```

Comment:
`startswith` will accept a tuple as the first item so you should be able to combine this check.